botocore/loaders.py

```diff
@@ -233,29 +307,27 @@ class Loader:
 
     FILE_LOADER_CLASS = JSONFileLoader
     # The included models in botocore/data/ that we ship with botocore.
-    BUILTIN_DATA_PATH = os.path.join(BOTOCORE_ROOT, 'data')
+    BUILTIN_DATA_PATH = BOTOCORE_ROOT.joinpath('data')
     # For convenience we automatically add ~/.aws/models to the data path.
-    CUSTOMER_DATA_PATH = os.path.join(
-        os.path.expanduser('~'), '.aws', 'models'
-    )
+    # this filepath can never be zipped and will always be on disk
+    CUSTOMER_DATA_PATH = pathlib.Path('~', '.aws', 'models').expanduser()
```

Comment:
This is subtly different since we're deferring expansion. That's likely not a problem, but it may warrant a spot check on at least Linux/macOS/Windows.

Reply:
would you rather i change it to `pathlib.Path('~').expanduser().joinpath('.aws', 'models')` or add a simple unit test?